class.jetpack.php

```diff
@@ -310,6 +310,10 @@ public static function init() {
 	 * Must never be called statically
 	 */
 	function plugin_upgrade() {
+		if( isset( $_GET['sync'] ) ) {
+			error_log('sync do action called ' );
+			do_action( 'jetpack_sync_all_registered_options' );
```

Comment:
Could you add the following before the action?

`/** This action is documented in class.jetpack.php */`